src/KafkaFlow.LogHandler.Microsoft/MicrosoftLogHandler.cs

```diff
@@ -15,21 +17,37 @@ public MicrosoftLogHandler(ILoggerFactory loggerFactory)
 
     public void Error(string message, Exception ex, object data)
     {
-        _logger.LogError(ex, "{Message} | Data: {Data}", message, JsonSerializer.Serialize(data));
+        _logger.LogError(ex, "{Message} | Data: {Data}", message, SerializeData(data));
     }
 
     public void Warning(string message, object data)
     {
-        _logger.LogWarning("{Message} | Data: {Data}", message, JsonSerializer.Serialize(data));
+        _logger.LogWarning("{Message} | Data: {Data}", message, SerializeData(data));
     }
 
     public void Info(string message, object data)
     {
-        _logger.LogInformation("{Message} | Data: {Data}", message, JsonSerializer.Serialize(data));
+        _logger.LogInformation("{Message} | Data: {Data}", message, SerializeData(data));
     }
 
     public void Verbose(string message, object data)
     {
-        _logger.LogDebug("{Message} | Data: {Data}", message, JsonSerializer.Serialize(data));
+        _logger.LogDebug("{Message} | Data: {Data}", message, SerializeData(data));
+    }
+
+    private string SerializeData(object data)
```

Comment:
The problem reported in the issue is not related to the exception, the error happens when there is an object that is not serializable (in the case of the issue is IntPtr).
If any type inside the data object is not serializable it will cause the error.
A new overload will not fix the problem here.

Reply:
We could have this overload also, it provides the best approach and promotes good practice. I didn't see those as mutually exclusive.

Reply:
But it will be a breaking change

Reply:
Its not replacing the current one but adding an overload accepting the exception as parameter. That isn't a breaking change.

Reply:
Also, with the current implementation if there is an exception while serializing it will discard the original log. This is probably not the best approach.

Reply:
It is a breaking change because we need to change the ILogHandler interface and it will break all projects with custom ILogHandlers.

IMO log should not causes errors in the application flow instead it is better to sinalize the problem without cause unnecesasry errors to the users.

Reply:
I agree with you that we should prevent breaking changes. But in this case, I think that is the best solution and although it is a breaking change, it's on compile time and should affect only a subset of users (the ones that are using the IlogHandler directly).

My opinion is based mostly on these 2 points:
- We shouldn't encourage passing complex objects to the log data context
- The exception shouldn't be passed inside the object data

With that in mind, I think it is justified to create that overload because I think it is the correct implementation. With that, in my opinion, the handling of the deserialization exception is unnecessary.

Reply:
I think it is a good compromise. We provide the correct way to log the exception in a warning while preventing the log serialization from raising an exception. Looks good to me.